fastlane/lib/fastlane/actions/app_store_build_number.rb

```diff
@@ -100,7 +100,7 @@ def self.order_versions(versions)
 
       def self.api_token(params)
         params[:api_key] ||= Actions.lane_context[SharedValues::APP_STORE_CONNECT_API_KEY]
-        api_token ||= Spaceship::ConnectAPI::Token.create(params[:api_key]) if params[:api_key]
+        api_token ||= Spaceship::ConnectAPI::Token.create(**params[:api_key]) if params[:api_key]
```

Comment:
@ainame What does this `**` do here? 🤔  Does his splat out the hash into keyword argments?

Reply:
> There is a minor breaking change in Ruby 3.0 which is called "Separation of positional and keyword arguments". In short, the below code won't run on Ruby 3.0. Previously you could pass a Hash object as keyword arguments to a method, but now you need to either use ** to turn Hash into keywords arguments or simply use keywords to pass parameters. This is the most common case we need to fix.

As I mentioned this is the biggest breaking change introduced in Ruby 3.0. 
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

> @ainame What does this ** do here? 🤔 Does his splat out the hash into keyword argments?

And the answer is YES! You can still use `Hash` for composing parameters in Ruby 3.0 but must need to use `**` to passing them into a method using keyword arguments. This works on Ruby 2.4 as well.

```
satoshinamai@xxx fastlane % rbenv local 2.4.10
satoshinamai@xxx fastlane % irb
irb(main):001:0> def hello(a: , b:)
irb(main):002:1>   p a, b
irb(main):003:1> end
=> :hello
irb(main):004:0> hello(**{ a: 1, b: 'ccc'})
1
"ccc"
=> [1, "ccc"]
irb(main):005:0> hello(**{ b: 1, a: 'ccc'})
"ccc"
1
=> ["ccc", 1]
```

---

Regarding this specific idiom `api_token ||= Spaceship::ConnectAPI::Token.create(**params[:api_key]) if params[:api_key]`

`params[:api_key]` must be a `Hash` thanks to `FastlaneCore::ConfigItem`'s validation and it should be the one composed by `app_store_connect_api_key` you added 😛